sound/soc/sof/core.c

```diff
@@ -503,6 +503,20 @@ int snd_sof_device_probe(struct device *dev, struct snd_sof_pdata *plat_data)
 }
 EXPORT_SYMBOL(snd_sof_device_probe);
 
+static int __force_rpm_active(struct device *device, void *data)
+{
+	pm_runtime_get_sync(device);
+	pm_runtime_forbid(device);
+	return 0;
+}
+
+static int __release_rpm(struct device *device, void *data)
+{
+	pm_runtime_put(device);
+	pm_runtime_allow(device);
```

Comment:
@kv2019i I think we should do allow() first and then put_autosuspend().

Reply:
Ack @ranj063 , I'll update.

Reply:
@ranj063 Now updated. It switched the order but decided to keep a simple put(). I think this makes more sense. We don't really know whether device is idle  as we only call get() to establish the forbid() blocker, and we are not really signaling whether the device is idle or not. I think put_autosuspend() would ignore the idle callback and this is not something we want. 

Makes sense?